File: app/src/main/java/com/checkin/app/checkin/Menu/MenuItemInteraction.java
package com.checkin.app.checkin.Menu;

import androidx.annotation.NonNull;

import com.checkin.app.checkin.Menu.Model.MenuItemModel;

public interface MenuItemInteraction {
    boolean onMenuItemAdded(@NonNull MenuItemModel item);

    boolean onMenuItemChanged(@NonNull MenuItemModel item, int count);

    void onMenuItemShowInfo(@NonNull MenuItemModel item);

    int getItemOrderedCount(@NonNull MenuItemModel item);
}

File: app/src/main/java/com/checkin/app/checkin/menu/listeners/MenuItemInteraction.kt
package com.checkin.app.checkin.menu.listeners

import com.checkin.app.checkin.menu.models.MenuItemModel

interface MenuItemInteraction {
    fun onMenuItemAdded(item: MenuItemModel): Boolean
    fun onMenuItemChanged(item: MenuItemModel, count: Int): Boolean
    fun onMenuItemShowInfo(item: MenuItemModel)
    fun getItemOrderedCount(item: MenuItemModel): Int
}